{
  "url": "https://api.github.com/repos/Consensys/teku/issues/7285",
  "repository_url": "https://api.github.com/repos/Consensys/teku",
  "labels_url": "https://api.github.com/repos/Consensys/teku/issues/7285/labels{/name}",
  "comments_url": "https://api.github.com/repos/Consensys/teku/issues/7285/comments",
  "events_url": "https://api.github.com/repos/Consensys/teku/issues/7285/events",
  "html_url": "https://github.com/Consensys/teku/issues/7285",
  "id": 1766923836,
  "node_id": "I_kwDOCM9I9M5pUR48",
  "number": 7285,
  "title": "Teku 23.6.0 having issues with GNOSIS chain",
  "user": {
    "login": "phd2018eth",
    "id": 121027629,
    "node_id": "U_kgDOBza8LQ",
    "avatar_url": "https://avatars.githubusercontent.com/u/121027629?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/phd2018eth",
    "html_url": "https://github.com/phd2018eth",
    "followers_url": "https://api.github.com/users/phd2018eth/followers",
    "following_url": "https://api.github.com/users/phd2018eth/following{/other_user}",
    "gists_url": "https://api.github.com/users/phd2018eth/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/phd2018eth/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/phd2018eth/subscriptions",
    "organizations_url": "https://api.github.com/users/phd2018eth/orgs",
    "repos_url": "https://api.github.com/users/phd2018eth/repos",
    "events_url": "https://api.github.com/users/phd2018eth/events{/privacy}",
    "received_events_url": "https://api.github.com/users/phd2018eth/received_events",
    "type": "User",
    "site_admin": false
  },
  "labels": [

  ],
  "state": "open",
  "locked": false,
  "assignee": null,
  "assignees": [

  ],
  "milestone": null,
  "comments": 12,
  "created_at": "2023-06-21T07:22:06Z",
  "updated_at": "2023-08-14T22:08:31Z",
  "closed_at": null,
  "author_association": "NONE",
  "active_lock_reason": null,
  "body": "<!-- \r\nBy filing an Issue, you are expected to comply with the Code of Conduct, \r\nincluding treating everyone with respect:\r\nhttps://github.com/ConsenSys/teku/blob/master/CODE-OF-CONDUCT.md\r\n\r\nNot all sections will apply to all issue types.\r\n-->\r\n\r\n### Description\r\n[Detailed description of the problem and the impact it has]\r\nSignificant reduction in effectiveness caused by periodical low counts (0-70%）of correct target and head.\r\nSometimes with \"ERROR - Refusing to sign attestation\" \r\n### Steps to Reproduce (Bug)\r\n[Please be as specific as possible]\r\nUpgrade TEKU from 23.4.0 to 23.6.0 using eth-docker stack.\r\nDowngrade back to 23.4.0 or 23.5.0 both behave normally.\r\n**Expected behavior:** [What you expect to happen]\r\nNo change in the validation process\r\n**Actual behavior:** [What actually happens]\r\nERROR log：[message.1.txt](https://github.com/ConsenSys/teku/files/11814721/message.1.txt)\r\n\r\n**Frequency:** [How regularly does it occur?]\r\nAlways with bad attestation performance but only log ERROR after running for a while (an hour also). \r\n### Versions (Add all that apply)\r\n* Software version: [`teku --version` or look for log lines starting with `Teku version:`] 23.6.0\r\n* Java version: [`java -version`] java-17\r\n* OS Name & Version: [`cat /etc/*release`] Ubuntu 22.04\r\n* Docker Version: [`docker version`] 24.0.2\r\n* Cloud VM, type, size: [Amazon Web Services I3-large] N/A\r\n* Nethermind version: 1.19.2\r\n* eth-docker version: 2.2.9.2",
  "closed_by": null,
  "reactions": {
    "url": "https://api.github.com/repos/Consensys/teku/issues/7285/reactions",
    "total_count": 0,
    "+1": 0,
    "-1": 0,
    "laugh": 0,
    "hooray": 0,
    "confused": 0,
    "heart": 0,
    "rocket": 0,
    "eyes": 0
  },
  "timeline_url": "https://api.github.com/repos/Consensys/teku/issues/7285/timeline",
  "performed_via_github_app": null,
  "state_reason": null
}
[
  {
    "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1600786132",
    "html_url": "https://github.com/Consensys/teku/issues/7285#issuecomment-1600786132",
    "issue_url": "https://api.github.com/repos/Consensys/teku/issues/7285",
    "id": 1600786132,
    "node_id": "IC_kwDOCM9I9M5fag7U",
    "user": {
      "login": "StefanBratanov",
      "id": 14827647,
      "node_id": "MDQ6VXNlcjE0ODI3NjQ3",
      "avatar_url": "https://avatars.githubusercontent.com/u/14827647?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/StefanBratanov",
      "html_url": "https://github.com/StefanBratanov",
      "followers_url": "https://api.github.com/users/StefanBratanov/followers",
      "following_url": "https://api.github.com/users/StefanBratanov/following{/other_user}",
      "gists_url": "https://api.github.com/users/StefanBratanov/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/StefanBratanov/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/StefanBratanov/subscriptions",
      "organizations_url": "https://api.github.com/users/StefanBratanov/orgs",
      "repos_url": "https://api.github.com/users/StefanBratanov/repos",
      "events_url": "https://api.github.com/users/StefanBratanov/events{/privacy}",
      "received_events_url": "https://api.github.com/users/StefanBratanov/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2023-06-21T12:56:29Z",
    "updated_at": "2023-06-21T12:56:29Z",
    "author_association": "CONTRIBUTOR",
    "body": "Hi, thanks for raising the issue. Will look into it. Can you confirm that downgrading helps and these issues are not there and when upgrading you see these issues periodically?",
    "reactions": {
      "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1600786132/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1601359427",
    "html_url": "https://github.com/Consensys/teku/issues/7285#issuecomment-1601359427",
    "issue_url": "https://api.github.com/repos/Consensys/teku/issues/7285",
    "id": 1601359427,
    "node_id": "IC_kwDOCM9I9M5fcs5D",
    "user": {
      "login": "zilm13",
      "id": 6196452,
      "node_id": "MDQ6VXNlcjYxOTY0NTI=",
      "avatar_url": "https://avatars.githubusercontent.com/u/6196452?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/zilm13",
      "html_url": "https://github.com/zilm13",
      "followers_url": "https://api.github.com/users/zilm13/followers",
      "following_url": "https://api.github.com/users/zilm13/following{/other_user}",
      "gists_url": "https://api.github.com/users/zilm13/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/zilm13/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/zilm13/subscriptions",
      "organizations_url": "https://api.github.com/users/zilm13/orgs",
      "repos_url": "https://api.github.com/users/zilm13/repos",
      "events_url": "https://api.github.com/users/zilm13/events{/privacy}",
      "received_events_url": "https://api.github.com/users/zilm13/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2023-06-21T18:19:30Z",
    "updated_at": "2023-06-21T18:19:30Z",
    "author_association": "CONTRIBUTOR",
    "body": "@phd2018eth could you please provide logs since the start of the epoch (since around 2023-06-16 16:02 your time) both for Teku and Nethermind. It may help in investigation",
    "reactions": {
      "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1601359427/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1613294117",
    "html_url": "https://github.com/Consensys/teku/issues/7285#issuecomment-1613294117",
    "issue_url": "https://api.github.com/repos/Consensys/teku/issues/7285",
    "id": 1613294117,
    "node_id": "IC_kwDOCM9I9M5gKOol",
    "user": {
      "login": "phd2018eth",
      "id": 121027629,
      "node_id": "U_kgDOBza8LQ",
      "avatar_url": "https://avatars.githubusercontent.com/u/121027629?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/phd2018eth",
      "html_url": "https://github.com/phd2018eth",
      "followers_url": "https://api.github.com/users/phd2018eth/followers",
      "following_url": "https://api.github.com/users/phd2018eth/following{/other_user}",
      "gists_url": "https://api.github.com/users/phd2018eth/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/phd2018eth/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/phd2018eth/subscriptions",
      "organizations_url": "https://api.github.com/users/phd2018eth/orgs",
      "repos_url": "https://api.github.com/users/phd2018eth/repos",
      "events_url": "https://api.github.com/users/phd2018eth/events{/privacy}",
      "received_events_url": "https://api.github.com/users/phd2018eth/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2023-06-29T14:34:11Z",
    "updated_at": "2023-06-29T14:34:11Z",
    "author_association": "NONE",
    "body": "> Hi, thanks for raising the issue. Will look into it. Can you confirm that downgrading helps and these issues are not there and when upgrading you see these issues periodically?\r\n\r\nconfirm",
    "reactions": {
      "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1613294117/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1613297987",
    "html_url": "https://github.com/Consensys/teku/issues/7285#issuecomment-1613297987",
    "issue_url": "https://api.github.com/repos/Consensys/teku/issues/7285",
    "id": 1613297987,
    "node_id": "IC_kwDOCM9I9M5gKPlD",
    "user": {
      "login": "phd2018eth",
      "id": 121027629,
      "node_id": "U_kgDOBza8LQ",
      "avatar_url": "https://avatars.githubusercontent.com/u/121027629?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/phd2018eth",
      "html_url": "https://github.com/phd2018eth",
      "followers_url": "https://api.github.com/users/phd2018eth/followers",
      "following_url": "https://api.github.com/users/phd2018eth/following{/other_user}",
      "gists_url": "https://api.github.com/users/phd2018eth/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/phd2018eth/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/phd2018eth/subscriptions",
      "organizations_url": "https://api.github.com/users/phd2018eth/orgs",
      "repos_url": "https://api.github.com/users/phd2018eth/repos",
      "events_url": "https://api.github.com/users/phd2018eth/events{/privacy}",
      "received_events_url": "https://api.github.com/users/phd2018eth/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2023-06-29T14:36:11Z",
    "updated_at": "2023-06-29T14:36:11Z",
    "author_association": "NONE",
    "body": "> @phd2018eth could you please provide logs since the start of the epoch (since around 2023-06-16 16:02 your time) both for Teku and Nethermind. It may help in investigation\r\n\r\nSorry, the older containers are removed when I relaunched the services. I only have that part of the logs saved. ",
    "reactions": {
      "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1613297987/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1617923107",
    "html_url": "https://github.com/Consensys/teku/issues/7285#issuecomment-1617923107",
    "issue_url": "https://api.github.com/repos/Consensys/teku/issues/7285",
    "id": 1617923107,
    "node_id": "IC_kwDOCM9I9M5gb4wj",
    "user": {
      "login": "StefanBratanov",
      "id": 14827647,
      "node_id": "MDQ6VXNlcjE0ODI3NjQ3",
      "avatar_url": "https://avatars.githubusercontent.com/u/14827647?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/StefanBratanov",
      "html_url": "https://github.com/StefanBratanov",
      "followers_url": "https://api.github.com/users/StefanBratanov/followers",
      "following_url": "https://api.github.com/users/StefanBratanov/following{/other_user}",
      "gists_url": "https://api.github.com/users/StefanBratanov/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/StefanBratanov/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/StefanBratanov/subscriptions",
      "organizations_url": "https://api.github.com/users/StefanBratanov/orgs",
      "repos_url": "https://api.github.com/users/StefanBratanov/repos",
      "events_url": "https://api.github.com/users/StefanBratanov/events{/privacy}",
      "received_events_url": "https://api.github.com/users/StefanBratanov/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2023-07-03T10:54:47Z",
    "updated_at": "2023-07-03T10:55:11Z",
    "author_association": "CONTRIBUTOR",
    "body": "Hi @phd2018eth , are you sure you are not running two validator clients simulatenously somehow when upgrading? Have you performed upgrades using `eth-docker` before?\r\n\r\n```\r\n2023-06-16 16:03:20.576 ERROR - Refusing to sign attestation:  source epoch (599224) is safe, target epoch (599225) is unsafe.  Previously signed source epoch 599222, target epoch 599225\r\n```\r\n\r\nThis error basically means you have already signed an attestation for a validator for the given target epoch. It is a definite no-no to do double signing, since this could result in slashing. ",
    "reactions": {
      "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1617923107/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1625244471",
    "html_url": "https://github.com/Consensys/teku/issues/7285#issuecomment-1625244471",
    "issue_url": "https://api.github.com/repos/Consensys/teku/issues/7285",
    "id": 1625244471,
    "node_id": "IC_kwDOCM9I9M5g30M3",
    "user": {
      "login": "StefanBratanov",
      "id": 14827647,
      "node_id": "MDQ6VXNlcjE0ODI3NjQ3",
      "avatar_url": "https://avatars.githubusercontent.com/u/14827647?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/StefanBratanov",
      "html_url": "https://github.com/StefanBratanov",
      "followers_url": "https://api.github.com/users/StefanBratanov/followers",
      "following_url": "https://api.github.com/users/StefanBratanov/following{/other_user}",
      "gists_url": "https://api.github.com/users/StefanBratanov/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/StefanBratanov/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/StefanBratanov/subscriptions",
      "organizations_url": "https://api.github.com/users/StefanBratanov/orgs",
      "repos_url": "https://api.github.com/users/StefanBratanov/repos",
      "events_url": "https://api.github.com/users/StefanBratanov/events{/privacy}",
      "received_events_url": "https://api.github.com/users/StefanBratanov/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2023-07-07T11:03:31Z",
    "updated_at": "2023-07-07T11:03:48Z",
    "author_association": "CONTRIBUTOR",
    "body": "Also @phd2018eth this could potentially happen in case of long reorgs or state regenerations when running low on resources. What `-Xmx` parameter you run Teku with? `6g` is advised for mainnet, but for Gnosis, it could be better to run with `7g` because of the faster slot times.",
    "reactions": {
      "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1625244471/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1628445596",
    "html_url": "https://github.com/Consensys/teku/issues/7285#issuecomment-1628445596",
    "issue_url": "https://api.github.com/repos/Consensys/teku/issues/7285",
    "id": 1628445596,
    "node_id": "IC_kwDOCM9I9M5hEBuc",
    "user": {
      "login": "phd2018eth",
      "id": 121027629,
      "node_id": "U_kgDOBza8LQ",
      "avatar_url": "https://avatars.githubusercontent.com/u/121027629?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/phd2018eth",
      "html_url": "https://github.com/phd2018eth",
      "followers_url": "https://api.github.com/users/phd2018eth/followers",
      "following_url": "https://api.github.com/users/phd2018eth/following{/other_user}",
      "gists_url": "https://api.github.com/users/phd2018eth/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/phd2018eth/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/phd2018eth/subscriptions",
      "organizations_url": "https://api.github.com/users/phd2018eth/orgs",
      "repos_url": "https://api.github.com/users/phd2018eth/repos",
      "events_url": "https://api.github.com/users/phd2018eth/events{/privacy}",
      "received_events_url": "https://api.github.com/users/phd2018eth/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2023-07-10T08:06:45Z",
    "updated_at": "2023-07-10T08:06:45Z",
    "author_association": "NONE",
    "body": "> Hi @phd2018eth , are you sure you are not running two validator clients simulatenously somehow when upgrading? Have you performed upgrades using `eth-docker` before?\r\n> \r\n> ```\r\n> 2023-06-16 16:03:20.576 ERROR - Refusing to sign attestation:  source epoch (599224) is safe, target epoch (599225) is unsafe.  Previously signed source epoch 599222, target epoch 599225\r\n> ```\r\n> \r\n> This error basically means you have already signed an attestation for a validator for the given target epoch. It is a definite no-no to do double signing, since this could result in slashing.\r\n\r\nNo question about it. It seemed failing to follow the chain then try to catch up then conflicts happen. As I have detailed in the original report, the exception won't happen right away after upgrading. It will run appearing normal initially, then attestation performance deteriorates for some time. Then the exception would occur after an hour.\r\n",
    "reactions": {
      "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1628445596/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1628445932",
    "html_url": "https://github.com/Consensys/teku/issues/7285#issuecomment-1628445932",
    "issue_url": "https://api.github.com/repos/Consensys/teku/issues/7285",
    "id": 1628445932,
    "node_id": "IC_kwDOCM9I9M5hEBzs",
    "user": {
      "login": "phd2018eth",
      "id": 121027629,
      "node_id": "U_kgDOBza8LQ",
      "avatar_url": "https://avatars.githubusercontent.com/u/121027629?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/phd2018eth",
      "html_url": "https://github.com/phd2018eth",
      "followers_url": "https://api.github.com/users/phd2018eth/followers",
      "following_url": "https://api.github.com/users/phd2018eth/following{/other_user}",
      "gists_url": "https://api.github.com/users/phd2018eth/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/phd2018eth/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/phd2018eth/subscriptions",
      "organizations_url": "https://api.github.com/users/phd2018eth/orgs",
      "repos_url": "https://api.github.com/users/phd2018eth/repos",
      "events_url": "https://api.github.com/users/phd2018eth/events{/privacy}",
      "received_events_url": "https://api.github.com/users/phd2018eth/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2023-07-10T08:07:00Z",
    "updated_at": "2023-07-10T08:08:32Z",
    "author_association": "NONE",
    "body": "> Also @phd2018eth this could potentially happen in case of long reorgs or state regenerations when running low on resources. What `-Xmx` parameter you run Teku with? `6g` is advised for mainnet, but for Gnosis, it could be better to run with `7g` because of the faster slot times.\r\n\r\n8g, but it's indeed a slow machine with an old 4-core cpu.  Does 23.6.0 require significantly more resources? \r\n",
    "reactions": {
      "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1628445932/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1628515014",
    "html_url": "https://github.com/Consensys/teku/issues/7285#issuecomment-1628515014",
    "issue_url": "https://api.github.com/repos/Consensys/teku/issues/7285",
    "id": 1628515014,
    "node_id": "IC_kwDOCM9I9M5hESrG",
    "user": {
      "login": "phd2018eth",
      "id": 121027629,
      "node_id": "U_kgDOBza8LQ",
      "avatar_url": "https://avatars.githubusercontent.com/u/121027629?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/phd2018eth",
      "html_url": "https://github.com/phd2018eth",
      "followers_url": "https://api.github.com/users/phd2018eth/followers",
      "following_url": "https://api.github.com/users/phd2018eth/following{/other_user}",
      "gists_url": "https://api.github.com/users/phd2018eth/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/phd2018eth/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/phd2018eth/subscriptions",
      "organizations_url": "https://api.github.com/users/phd2018eth/orgs",
      "repos_url": "https://api.github.com/users/phd2018eth/repos",
      "events_url": "https://api.github.com/users/phd2018eth/events{/privacy}",
      "received_events_url": "https://api.github.com/users/phd2018eth/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2023-07-10T08:45:32Z",
    "updated_at": "2023-07-10T09:03:15Z",
    "author_association": "NONE",
    "body": "Just tried to upgrade to 23.6.1 with -Xmx10g. This time the exception appeared earlier on, but then recovered and works fine so far.  Then later on again after some empty blocks. The complete log file is attached.\r\n \r\n[gnosis_teku23.6.1.log](https://github.com/Consensys/teku/files/11999901/gnosis_teku23.6.1.log)\r\n",
    "reactions": {
      "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1628515014/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1628534923",
    "html_url": "https://github.com/Consensys/teku/issues/7285#issuecomment-1628534923",
    "issue_url": "https://api.github.com/repos/Consensys/teku/issues/7285",
    "id": 1628534923,
    "node_id": "IC_kwDOCM9I9M5hEXiL",
    "user": {
      "login": "StefanBratanov",
      "id": 14827647,
      "node_id": "MDQ6VXNlcjE0ODI3NjQ3",
      "avatar_url": "https://avatars.githubusercontent.com/u/14827647?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/StefanBratanov",
      "html_url": "https://github.com/StefanBratanov",
      "followers_url": "https://api.github.com/users/StefanBratanov/followers",
      "following_url": "https://api.github.com/users/StefanBratanov/following{/other_user}",
      "gists_url": "https://api.github.com/users/StefanBratanov/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/StefanBratanov/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/StefanBratanov/subscriptions",
      "organizations_url": "https://api.github.com/users/StefanBratanov/orgs",
      "repos_url": "https://api.github.com/users/StefanBratanov/repos",
      "events_url": "https://api.github.com/users/StefanBratanov/events{/privacy}",
      "received_events_url": "https://api.github.com/users/StefanBratanov/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2023-07-10T08:58:50Z",
    "updated_at": "2023-07-10T08:58:50Z",
    "author_association": "CONTRIBUTOR",
    "body": "> Just tried to upgrade to 23.6.1 with -Xmx10g. This time the exception appeared earlier on, but then recovered and works fine so far. The complete log file is attached. [gnosis_teku23.6.1.log](https://github.com/Consensys/teku/files/11999674/gnosis_teku23.6.1.log)\r\n\r\nThank you. I can see one of the errors `java.lang.IllegalArgumentException: Committee information must be derived from a state no older than the previous epoch.` comes from a code path that was changed, so will see what could be the effect.",
    "reactions": {
      "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1628534923/reactions",
      "total_count": 1,
      "+1": 1,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1628543346",
    "html_url": "https://github.com/Consensys/teku/issues/7285#issuecomment-1628543346",
    "issue_url": "https://api.github.com/repos/Consensys/teku/issues/7285",
    "id": 1628543346,
    "node_id": "IC_kwDOCM9I9M5hEZly",
    "user": {
      "login": "phd2018eth",
      "id": 121027629,
      "node_id": "U_kgDOBza8LQ",
      "avatar_url": "https://avatars.githubusercontent.com/u/121027629?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/phd2018eth",
      "html_url": "https://github.com/phd2018eth",
      "followers_url": "https://api.github.com/users/phd2018eth/followers",
      "following_url": "https://api.github.com/users/phd2018eth/following{/other_user}",
      "gists_url": "https://api.github.com/users/phd2018eth/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/phd2018eth/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/phd2018eth/subscriptions",
      "organizations_url": "https://api.github.com/users/phd2018eth/orgs",
      "repos_url": "https://api.github.com/users/phd2018eth/repos",
      "events_url": "https://api.github.com/users/phd2018eth/events{/privacy}",
      "received_events_url": "https://api.github.com/users/phd2018eth/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2023-07-10T09:04:30Z",
    "updated_at": "2023-07-10T09:10:38Z",
    "author_association": "NONE",
    "body": "Just updated with more logs with another exception later on.  But unlike the previous run with 23.6.0 and 8g heap, it is able to recover and perform decent, so I will keep it running in this mode. Thanks for looking into it.",
    "reactions": {
      "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1628543346/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1678144235",
    "html_url": "https://github.com/Consensys/teku/issues/7285#issuecomment-1678144235",
    "issue_url": "https://api.github.com/repos/Consensys/teku/issues/7285",
    "id": 1678144235,
    "node_id": "IC_kwDOCM9I9M5kBnLr",
    "user": {
      "login": "rolfyone",
      "id": 2967240,
      "node_id": "MDQ6VXNlcjI5NjcyNDA=",
      "avatar_url": "https://avatars.githubusercontent.com/u/2967240?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/rolfyone",
      "html_url": "https://github.com/rolfyone",
      "followers_url": "https://api.github.com/users/rolfyone/followers",
      "following_url": "https://api.github.com/users/rolfyone/following{/other_user}",
      "gists_url": "https://api.github.com/users/rolfyone/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/rolfyone/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/rolfyone/subscriptions",
      "organizations_url": "https://api.github.com/users/rolfyone/orgs",
      "repos_url": "https://api.github.com/users/rolfyone/repos",
      "events_url": "https://api.github.com/users/rolfyone/events{/privacy}",
      "received_events_url": "https://api.github.com/users/rolfyone/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2023-08-14T22:08:30Z",
    "updated_at": "2023-08-14T22:08:30Z",
    "author_association": "CONTRIBUTOR",
    "body": "how is Gnosis on 23.8.0? It should improve further...",
    "reactions": {
      "url": "https://api.github.com/repos/Consensys/teku/issues/comments/1678144235/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  }
]
